src/facilities-context/reducer.tsx

```diff
@@ -0,0 +1,37 @@
+import { Facility } from "../page-multi-facility/types";
+import { FacilitiesState } from "./FacilitiesContext";
+import * as actions from "./actions";
+
+export function facilitiesReducer(state: FacilitiesState, action: actions.FacilitiesActions): FacilitiesState {
+  switch (action.type) {
+    case actions.REQUEST_FACILITIES:
+      return Object.assign({}, state, { loading: true });
+    case actions.RECEIVE_FACILITIES:
+      return Object.assign({}, state, {
+        loading: false,
+        facilities: action.payload
```

Comment:
I went back and forth here with passing the full state updates to the dispatch payload, or defining them here in the case statement. Does anyone have a preference?